pkg/distributor/distributor.go

```diff
@@ -168,8 +168,9 @@ type Config struct {
 	// this (and should never use it) but this feature is used by other projects built on top of it
 	SkipLabelNameValidation bool `yaml:"-"`
 
-	// This config is dynamically injected because it is defined in the querier config.
-	ShuffleShardingLookbackPeriod time.Duration `yaml:"-"`
+	// The two fields below are dynamically injected from the querier and ingester configs
+	ShuffleShardingIngestersEnabled bool          `yaml:"-"`
```

Comment:
This name is too generic and it may be confusing -- it looks like it may affect write path too. Let's give it more specific name. Eg. `ShuffleShardingIngestersOnReadPathEnabled`. (Or something better)

Alternatively we can keep the original logic and simply set `ShuffleShardingLookbackPeriod` to 0 if shuffle sharding of ingesters on read path is not enabled.

Reply:
Updated - thanks for the suggestion, much cleaner this way :)